### ISSUE TITLE
Atom Editor freezes occasionally

### ISSUE THREAD

**Issue by diegoferigo** [jamesnetherton/docker-atom-editor#91]
@jamesnetherton I'm trying to debug https://github.com/diegoferigo/dockerfiles/issues/12 and I tested the behavior of your image. I have the same freezes described in the linked issue, but the error generated now is:

```
docker run -it \
           -v /tmp/.X11-unix/:/tmp/.X11-unix/ \
           -v /dev/shm:/dev/shm \
           -v ${HOME}/.atom:/home/atom/.atom \
           -e DISPLAY \
           jamesnetherton/docker-atom-editor

[...]
[18:0802/141212:ERROR:browser_gpu_channel_host_factory.cc(119)] Failed to launch GPU process.
```

Including in the image the packages for HW acceleration (`libgl1-mesa-glx` `libgl1-mesa-dri`) the error becomes exactly the same I got:

```
[20:0802/142316:WARNING:x11_util.cc(1406)] X error received: serial 5126, error_code 3 (BadWindow (invalid Window parameter)), request_code 3, minor_code 0 (X_GetWindowAttributes)
```

Did you ever notice something similar? It looks like there's something going on between X11 and KF5/Plasma.

**Comment by jamesnetherton** [comment]
I'll have to experiment to see if I can reproduce it. It's not something that I have come across in the past.

But, I've moved to VS Code so I don't use Atom too often these days.

Does it happen frequently? Or just randomly over a long period?

**Comment by diegoferigo** [comment]
I experimented VS Code as well, but unfortunately the C++ support (debugging excluded) is still not as good as in Atom, which has so far a decent support of CMake + clang :disappointed: 

I noticed this on two setups, the first with Arch Linux (rolling) and the second with Ubuntu 17.04, both with KF5/Plasma. My old setup, running on Ubuntu 16.10 (or 04, don't remember exactly) never had such issues, hence I suppose this is a regression of the most recent X11 releases. If this is the case unfortunately it would take a lot of time to debug, I'd wanted to ask you since you're maintaining this Atom image.

It happens every time and right after the container is created. The easier way to trigger it is opening the menu and browsing the items with the cursor. The editor freezes for a couple of seconds (the host system as well) and then the errors I reported are displayed. It keeps happening doing the same operation again, even though other actions can trigger the freezes e.g. opening folders in the tree view. For the sake of clarity I didn't created the container with `-d`, using `-it` the output of Atom can be directly read from the stdout of the terminal.

**Comment by jamesnetherton** [comment]
I'm running Fedora 25 with i3. I get the `Failed to launch GPU process` message, but there's no freezing. Adding the `libgl1-mesa-glx libgl1-mesa-dri` packages fixes up the error.

I'll try to replicate your environment and see what happens.

**Comment by diegoferigo** [comment]
Good to know that at least you manage to replicate that. My current setup is a Dell AIO with an `i5-7500` CPU and basically just a plain Ubuntu 17.04 with `kubuntu-desktop` and `docker` from the ppa, 2 days fresh. However, I remember noticing this issue in my laptop (Dell XPS 13 9343) running Arch Linux at least 4 / 5 months ago.

**Comment by jamesnetherton** [comment]
I can't replicate your hardware but I tested in a Kubuntu 17.04 VM. I still see the errors you mentioned earlier in the console output, but there is no freezing.

The menus open fine and everything seems to run smoothly.

**Comment by diegoferigo** [comment]
Ok so it doesn't happen in a virtualized environment, and I can tell you it doesn't happen either in a regular `ubuntu-session` with unity on the same setup previously described, tested this setup either. It looks like KF5 / Plasma on real hardware messes things up, might it be `kwin` related? It's the only big difference between the DEs I see here.

**Comment by jamesnetherton** [comment]
Yes, could be related to `kwin`. Maybe try to obtain some [debug logs](https://community.kde.org/KWin/Debugging)?

**Comment by mitsest** [comment]
I 'm also experiencing this behaviour.
I 'm under ubuntu 16.04

**Comment by jamesnetherton** [comment]
@mitsest can you give some more details about your environment? 

What window manager do you run, etc.

**Comment by mitsest** [comment]
Sure!

I 'm using unity and here is the lspci output

mits@mits:~/nativescript/dev_env (master)*$ lspci | grep -i --color 'vga\|3d\|2d'

00:02.0 VGA compatible controller: Intel Corporation 3rd Gen Core processor Graphics Controller (rev 09)

**Comment by jamesnetherton** [comment]
On Fedora I have:

```
00:02.0 VGA compatible controller: Intel Corporation 4th Gen Core Processor Integrated Graphics Controller (rev 06)`
```
I don't experience any problems. Maybe I'll try to build a Ubuntu VM and see if I can reproduce.

**Comment by mitsest** [comment]
OK, cheers man!
I 'll wait to hear about your test results.

Thanks again

**Comment by mitsest** [comment]
I think it happens when it tries to load the welcome screen. 
I 'm trying to apm disable welcome inside the DockerFile but it fails with message:

`Could not find config.cson. Run Atom first?`

So, in order for this file to appear, you have to run atom first? What a load of bull@@

**Comment by jamesnetherton** [comment]
Can you not generate config.cson locally (or in another container) and just add it to your image?

**Comment by mitsest** [comment]
Yep, you beat me to it! Just tried it. 
I tried with a minimum cson file and apm disable welcome, but the freeze persists :/

**Comment by mitsest** [comment]
After some tests on Debian I don't think it's easily reproducible.
But gnome3 seems to freeze less...

**Comment by diegoferigo** [comment]
> [!NOTE]  
> Cleaning up old issues and pull requests I opened over the past years that are now outdated or stale. If this is still relevant, feel free to reopen it or create a new one with updated details. Thanks!
